src/app/lib/getTags.ts

```diff
@@ -0,0 +1,78 @@
+import type {
+  BlogArticleData,
+  StrapiResponse,
+  TagAttributes,
+} from '@/types/strapi';
+import { TagStrapiApi } from '@/utils/strapi/StrapiApi';
+
+export interface GetTagsResponse extends StrapiResponse<TagAttributes> {
+  url: string;
+}
+
+export async function getTags(): Promise<GetTagsResponse> {
+  const urlParams = new TagStrapiApi().addPaginationParams({
+    page: 1,
+    pageSize: 20,
+    withCount: false,
+  });
+  const apiBaseUrl = urlParams.getApiBaseUrl();
+  const apiUrl = urlParams.getApiUrl();
+  const accessToken = urlParams.getApiAccessToken();
+  const res = await fetch(decodeURIComponent(apiUrl), {
+    cache: 'force-cache',
+    headers: {
+      Authorization: `Bearer ${accessToken}`,
+    },
+  });
+
+  if (!res.ok) {
+    throw new Error('Failed to fetch data');
+  }
+
+  const data = await res.json().then((output) => {
+    const filteredData = output.data;
+
+    return {
+      meta: output.meta,
+      data: filteredData?.sort(
```

Comment:
As mentionned in dm, this seems to be a duplication within https://github.com/jumperexchange/jumper-exchange/pull/1304/files#diff-39e2e3ff141d934ff49f20712a6e7c4bf32edcbf503ab3a9bc28e0fdd9e89fc3R17

Reply:
Hey @oktapodia !

Thanks for the reminder, so in order to achieve the expected outcome, I do need to sort twice unless you have an idea on how to refactor?

Sorting
1.) sort tags by publishedDate of blog_articles which is the relational collection of tags
2.) sort tags depending on `const predefinedOrder = ['Announcement', 'Partner', 'Bridge'];` which is requested by marketing and this solution was suggested by @tcheee 

Other than that, I noticed that parts of sorting took place way too late and could be handled in `getTags.ts` as well! I tried refactoring it, please give it one more look  💛